clang/lib/AST/Expr.cpp

```diff
@@ -1615,22 +1615,24 @@ QualType CallExpr::getCallReturnType(const ASTContext &Ctx) const {
   return FnType->getReturnType();
 }
 
-const Attr *CallExpr::getUnusedResultAttr(const ASTContext &Ctx) const {
+std::pair<const NamedDecl *, const Attr *>
+CallExpr::getUnusedResultAttr(const ASTContext &Ctx) const {
+  // If the callee is marked nodiscard, return that attribute
+  const Decl *D = getCalleeDecl();
+  if (const auto *A = D->getAttr<WarnUnusedResultAttr>())
```

Comment:
This looks like an unconditional access of `D` but it is not obvious this is a correct assumption

Reply:
Indeed, it seems I forgot to add a check for `nullptr`.
Will open a PR to fix this later today.